app/adapters/addon-service.ts

```diff
@@ -9,10 +11,25 @@ export const addonServiceAPIUrl = `${addonServiceUrl}${addonServiceNamespace}/`;
 export default class AddonServiceAdapter extends JSONAPIAdapter {
     host = addonServiceUrl.replace(/\/$/, ''); // Remove trailing slash to avoid // in URLs
     namespace = addonServiceNamespace;
+    @service currentUser!: CurrentUser;
 
     ajaxOptions(url: string, type: string, options?: any): object {
         const _ajaxopts: any = super.ajaxOptions(url, type, options);
         _ajaxopts.credentials = 'include';
         return _ajaxopts;
     }
+
+    ajax(url, method, options) {
+        return super.ajax(method === 'GET' ? this.transformUrl(url) : url, method, options);
+    }
+
+    transformUrl(url: string): string {
+        const { viewOnlyToken } = this.currentUser;
+        if (!viewOnlyToken) {
+            return url;
+        }
+        const separator = url.includes('?') ? '&' : '?';
+        return `${url}${separator}view_only=${viewOnlyToken}`;
```

Comment:
Slightly more stable way of setting the view_only param may be to use the JS URL interface
```
const returnUrl = new URL(url!);
returnUrl.searchParams.set('view_only', viewOnlyToken);
return returnUrl.toString();
```